File: internal/codeintel/context/chunkers/testdata/basic.go
// Package main docstring
package main

import "fmt"

// globalVarAlone docstring
var globalVarAlone = "bar"

var (
	// globalVarInGroup docstring
	globalVarInGroup = "foo"
	globalVarInGroup2 = "foo"
)

// globalConstAlone docstring
const globalConstAlone = 0

const (
	// globalConstInGroup docstring
	globalConstInGroup = 1
	globalConstInGroup2 = 1
)

// Foo docstring
type Foo struct {
	// Foo.Field docstring
	Field string
}

// Foo.Do() docstring
func (f *Foo) Do() {
	fmt.Println("Do")
}

// Foo.DoNonPointer() docstring
func (f Foo) DoNonPointer() {
	fmt.Println("DoNonPointer")
}

type Foo2 struct {
	Field2 string
}

func (f *Foo2) Do() {
	fmt.Println("Do")
}

func (f Foo2) DoNonPointer() {
	fmt.Println("DoNonPointer")
}

// Bar docstring
type Bar interface {
	// Baz docstring
	Baz()
	Baz2()
}

type Bar2 interface{}

type GenericBarContainer[T Bar2] struct {
	Bars T[]
}

func (b *GenericBarContainer[T]) barContainerDo() {
}

// main docstring
func main() {
	// inline comment
	fmt.Println("Hello world")
}

func main2() {}

File: internal/codeintel/context/chunkers/testdata/basic.java
/*
 * This is a package doc
 */
package org.apache.maven.api.settings;

import java.util.Comparator;

/**
 *  This is a javadoc
 **/
public class Application {
    public static void main(String[] args) {
        System.out.println("Hello, World");
    }
}

// Inline comment for FooClass
public class FooClass {
	// Inline comment for field bar
	private int barField;

	/**
	 * Constructor for Foo
	 *
	 * Multiline comment
	 */
	public FooClass(int bar) {
		String baz = bar;
		this.barField = baz;
		this.fooMethod();
	}

	// Non-block comment for fooMethod
	public int fooMethod() {
		int n = 1;
		System.out.println("doing foo");
		return n;
	}

	class InnerFoo {
		private int innerFooField = 123;

		public static void innerFooStaticMethod() {
		}
	}

	private static final Comparator ANON_COMPARATOR = new Comparator() {
		public int compare(Object aObj1, Object aObj2) {
			return -1;
		}
	};
}

/**
 * Block comment for FooInterface
 */
public interface FooInterface extends Serializable {
	FooClass getFoo();
}
